projects/plugins/jetpack/modules/subscriptions/subscribe-overlay/class-jetpack-subscribe-overlay.php

```diff
@@ -0,0 +1,229 @@
+	 *
+	 * @return WP_Block_Template|null
+	 */
+	public function get_block_template_filter( $block_template, $id, $template_type ) {
+		if ( empty( $block_template ) && $template_type === 'wp_template_part' ) {
+			if ( $id === self::get_block_template_part_id() ) {
+				return $this->get_template();
+			}
+		}
+
+		return $block_template;
+	}
+
+	/**
+	 * Returns a custom template for the Subscribe Overlay.
+	 *
+	 * @return WP_Block_Template
+	 */
+	public function get_template() {
+		$template                 = new WP_Block_Template();
+		$template->theme          = get_stylesheet();
+		$template->slug           = self::BLOCK_TEMPLATE_PART_SLUG;
+		$template->id             = self::get_block_template_part_id();
+		$template->area           = 'uncategorized';
+		$template->content        = $this->get_subscribe_overlay_template_content();
+		$template->source         = 'plugin';
+		$template->type           = 'wp_template_part';
+		$template->title          = __( 'Jetpack Subscribe overlay', 'jetpack' );
+		$template->status         = 'publish';
+		$template->has_theme_file = false;
+		$template->is_custom      = true;
+		$template->description    = __( 'An overlay that shows up when someone visits your site', 'jetpack' );
+
+		return $template;
+	}
+
+	/**
+	 * Returns the initial content of the Subscribe Overlay template.
+	 * This can then be edited by the user.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_overlay_template_content() {
+		$site_name        = get_bloginfo( 'name' );
+		$site_description = get_bloginfo( 'description' ); // TODO can be empty, how to handle?
+		$skip_to_content  = __( 'Skip to content', 'jetpack' );
+
+		return <<<HTML
+	<!-- wp:group {"layout":{"type":"constrained","contentSize":"400px"}} -->
+	<div class="wp-block-group">
+		<!-- wp:site-logo {"width":90,"shouldSyncIcon":true,"align":"center","className":"is-style-rounded"} /-->
+	
+		<!-- wp:heading {"textAlign":"center","level":1} -->
+		<h1 class="wp-block-heading has-text-align-center">$site_name</h1>
+		<!-- /wp:heading -->
+
+		<!-- wp:paragraph {"align":"center"} -->
+		<p class='has-text-align-center'>$site_description</p>
+		<!-- /wp:paragraph -->
+
+		<!-- wp:jetpack/subscriptions /-->
+
+		<!-- wp:paragraph {"align":"center","className":"jetpack-subscribe-overlay__to-content"} -->
+		<p class="has-text-align-center jetpack-subscribe-overlay__to-content"><a href="#">$skip_to_content</a> ↓</p>
+		<!-- /wp:paragraph -->
+	</div>
+	<!-- /wp:group -->
+HTML;
+	}
+
+	/**
+	 * Enqueues JS to load overlay.
+	 *
+	 * @return void
+	 */
+	public function enqueue_assets() {
+		if ( $this->should_user_see_overlay() ) {
+			wp_enqueue_style( 'subscribe-overlay-css', plugins_url( 'subscribe-overlay.css', __FILE__ ), array(), JETPACK__VERSION );
+			wp_enqueue_script( 'subscribe-overlay-js', plugins_url( 'subscribe-overlay.js', __FILE__ ), array( 'wp-dom-ready' ), JETPACK__VERSION, true );
```

Comment:
Wondering if we should add the subscriptions module to webpack, build the frontend JS as well and include the JS from `jetpack/projects/plugins/jetpack/_inc/build`? 

Would help with minification and also allow importing from `@wordpress/dom-ready` directly, and build pipeline "externalizes" WP scripts instead of including them in bundles.

No strong opinion and current way is fine too.

Reply:
Seems like an idea for a separate PR.

Reply:
Yep sounds good 👍